docs/tutorials/talk-to-kafka.rst

```diff
@@ -0,0 +1,715 @@
+        def __init__(self, name, age):
+            self.name = name
+            self.age = age
+
+    # Define a class for the Key object
+    class Key(object):
+        def __init__(self, id):
+            self.id = id
+
+    # Converting dict to User object for AvroDeserializer function
+    def dict_to_user(obj, ctx):
+        return User(name=obj['name'],
+                    age=obj['age']
+        )
+
+    # Converting dict to Key object for AvroDeserializer function
+    def dict_to_key(obj, ctx):
+        return Key(id=obj["id"])
+
+    # Create the Avro deserializer
+    avro_deserializer_value = AvroDeserializer(
+        schema_registry_client, value_str, dict_to_user
+    )
+
+    avro_deserializer_key = AvroDeserializer(
+        schema_registry_client, key_str, dict_to_key
+    )
+
+    # Create the Kafka consumer
+    consumer = DeserializingConsumer(
+        conf
+    )
+
+    # Subscribe to the Kafka topic
+    consumer.subscribe(['demo-topic'])
+
+    # Consume messages from the Kafka topic
+    while True:
+        try:
+            msg = consumer.poll(1.0)
+
+            if msg is None:
+                continue
+
+            if msg.error():
+                print("Consumer error: {}".format(msg.error()))
+                continue
+            
+            user = avro_deserializer_value(
+                msg.value(), SerializationContext(msg.topic(), MessageField.VALUE)
+            )
+            key = avro_deserializer_key(
+                msg.key(), SerializationContext(msg.topic(), MessageField.KEY)
+            )
+
+            if user is not None and key is not None:
+                print(
+                    "Key --> {}\n"
+                    "User record --> name: {},"
+                    " age: {}\n".format(key.id, user.name, user.age)
+                )
+
+        except KeyboardInterrupt:
+                break
+
+        # Commit the offset for the message
+        consumer.commit()
+
+Observation
+~~~~~~~~~~~
+
+Once you execute the consumer program, you should see the following output on the second terminal while the consumer program stays running until you hit ``ctrl+C``:
+
+.. code:: shell
+
+    Key --> 1
+    User record --> name: John, age: 30
+
+Here's an overview of what each program does:
+
+Producer program
+""""""""""""""""""
+
+The producer program uses the ``confluent-kafka`` library to create an AvroProducer instance and send Avro messages to a Kafka broker. Here's a breakdown of what the program does:
+
+- Import the necessary libraries: ``confluent_kafka``, ``confluent_kafka.avro``, ``ssl``, and ``io``.
+- Define the URL of the Kafka broker and the Avro schema for the message.
+- Set up SSL configuration by creating an SSL context and loading the client certificate, key, and CA certificate.
+- Define the AvroProducer configuration, including the SSL settings and the Avro schema.
+- Create an AvroProducer instance.
+- Send a sample Avro message to the Kafka broker.
+
+Consumer program
```

Comment:
Similar comment – put this content in the section you actually create the consumer in.

Reply:
Same response https://github.com/aiven/devportal/pull/1733#discussion_r1136459492